source/common/filesystem/posix/filesystem_impl.cc

```diff
@@ -46,7 +46,14 @@ Api::IoCallBoolResult FileImplPosix::open(FlagSet in) {
   if (isOpen()) {
     return resultSuccess(true);
   }
-
+  if (destinationType() == DestinationType::Stdout) {
```

Comment:
To make this clearer to reviewers, since looking at *just* this code it looks bizarre that it ever worked - what happens is we get here with `path() == "/dev/stdout"`, which goes through the regular file-opening route below, which works in *most* environments.

With a missing `/dev/stdout` that file open fails.